url-detector/src/main/java/com/linkedin/urls/detection/DomainNameReader.java

```diff
@@ -320,6 +320,17 @@ public ReaderNextState readDomainName() {
 
     //while not done and not end of string keep reading.
     boolean done = false;
+    boolean isAllHexSoFar = _reader.canReadChars(3) &&
```

Comment:
This is similar logic to what was done on ```readCurrent``` method.
IP in HEXA should start with ```0x```